### PR TITLE
Move configuraiton of a few tools to pyproject.toml

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
* Move configuraiton of a few tools to pyproject.toml
* Unpinned `black`
* Reformatted by later version of `black`

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
As described by https://github.com/flyteorg/flyte/issues/859, some configuration could be consolidated. Stop pinning `black` to move forward using later version of it.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/859

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by honnix** [comment]
@wild-endeavor PTAL at this minor change. Thanks.

File: requirements-spark3.txt [hunk 1]
```diff
@@ -25,15 +25,15 @@ backcall==0.2.0
     # via ipython
 bcrypt==3.2.0
     # via paramiko
-black==19.10b0
+black==19.10b0; extra == "all-spark3"
```
Comment (wild-endeavor) [inline requirements-spark3.txt:28]: Oh so you're running into this too.  do you know how to fix this?  I've spent at least a few hours already looking into this.  For me, it usually adds `extra == "all"` in the regular file.  Somehow, our internal qualifier is getting appended onto black's.  It makes no sense to me.

Comment (wild-endeavor) [reply requirements-spark3.txt:28]: maybe @jeevb & @cosmicBboy too if you've seen this... basically running `make requirements` adds this "extra" thing to black (only ever to black, nothing else), which then causes pip to fail because this has nothing to do with black.  "all" and "all-spark3" are flytekit concepts.

Comment (honnix) [reply requirements-spark3.txt:28]: I see. I didn't notice this. Will look into it. Thanks for pointing out.

Comment (honnix) [reply requirements-spark3.txt]: It seems to happen "randomly" in any compiled file, and sometimes it doesn't happen at all. :thinking:

Comment (wild-endeavor) [reply requirements-spark3.txt]: yeah i've had the same experience.

Comment (honnix) [reply requirements-spark3.txt]: After debugging deeply into pip-compile and pip, I think pip-compile got confused by `blacked` introduced by both `papermill` and `flytekit` itself as part of notebook extra. pip-compile does a few rounds of resolution until nothing changed after a round, but the order of things appearing may result differently, so sometimes it choose the one without extra marker but sometimes it doesn't.

It's weird that papermill requires black as installation requirement. Maybe we can ask the maintainer about it.

On the hand, when the extra marker gets added to black, how does pip fail in your case? To me, when there is this marker, it is as if black were not added to requirements.txt when doing `pip install -r` because I'm not providing an extra, so it doesn't fail but may happily install a newer version of black.

Maybe to avoid dealing with black and having to pin the version. Let's just unpin it and do a big reformatting PR? It doesn't change too much as I tried.

Comment (honnix) [reply requirements-spark3.txt]: Hmm, it actually has real code dependency on black https://github.com/nteract/papermill/blob/ca46ca6756ff15c1d03857af911f8060e35da8ab/papermill/translators.py#L189

Comment (wild-endeavor) [reply requirements-spark3.txt]: Down to do a reformatting yeah.  Thanks for investigating!  Do you mind putting it in?  My failure scenario... I don't remember actually.  I pulled this branch and ran `make requirements` and it ran fine.  If I see it again before the reformat I'll add it here.

**Comment by wild-endeavor** [comment]
Hang on though... why the removal of spark3?  I think we still want to keep that right?  Like for the next few versions we still want the old API to function properly including the old spark tasks.

**Comment by honnix** [comment]
Not removal actually. `all` has been spark3 for some time, so we have been running CI only for spark3. I changed CI to run for both spark3 and 2. I think it's better to do in a separated PR though.

**Comment by honnix** [comment]
@wild-endeavor PTAL again. The failed codecov is unfortunate but doesn't seem reasonable to fix in this PR.

**Review by wild-endeavor — APPROVED** [review]

**Comment by honnix** [comment]
Now I feel sorry for all existing PRs. But shouldn't be too messy to resolve.
